jina/drivers/segment.py

```diff
@@ -54,8 +54,9 @@ def _apply_all(self, docs: 'DocumentSet', *args, **kwargs):
 
     @staticmethod
     def _add_chunks(doc, chunks):
+        num_siblings = len(chunks)
```

Comment:
I wonder if this way will get the number of chunks for this level? Because num_siblings is called in a for loop, the `docs_chunks` is from self.exec_fn(contents) (the segment() from executor attached). so `docs_chunks` is a list of chunks or batches of chunks.

What if we put this `num_siblings = len(chunks)` before the for loop (line 52)? Which best lines with what we want?

Reply:
What is problem do you want to solve at this point? Getting access to `len(doc.chunks)`?

Reply:
I think my understanding was wrong. It would be better to add an integration test to validate the value is what we expect.